### ISSUE TITLE
"Command Pattern" class causes unexpected error: Too many arguments

### ISSUE THREAD

**Issue by Lordnibbler** [python/mypy#3807]
I have a few command pattern classes, with a structure like this:

```python
class Command(object):
    @classmethod
    def run(cls, string: str):
        cls('foo')

    def __init__(self, string: str) -> None:
        pass
```

The idea here is a clean class with a single class method to be invoked, ie. `Command.run('foo')`.

Since adding mypy 0.521, I've noticed these errors pop up:

```
mypy ./command.py
command.py:4 error: Too many arguments for "Command" 
```

This can be resolved by placing the `__init__` method before the `run` method, but I think one of two things should be done here:

1. Make this error message clearer, explaining that an override of `__init__` must come before a method that invokes it
1. Ensure mypy correctly resolves the fact that `__init__` is overridden here, and don't print this error.

Thoughts?

**Comment by ilevkivskyi** [comment]
Thanks, I think this is a bug (false positive). But it looks like a duplicate of #1727. I am closing this one, and will instead raise priority of #1727 to normal.
